### PR TITLE
Graphql apollo client

### PR DESCRIPTION
**Pull request by Kasper-S-Jensen**
Created Apollo graphQl client.
Able to retrive list of countries and specif country
Fetched data is mapped to a composable list that display the country data.
We will need to change the endpoint and map to events instead, but functionality remains the same
Skærmbillede 2023-10-08 120547

### CODE REVIEW

**Review by Kutaiba-Nezar-Kashmar — APPROVED** [review]
🔥

File: app/build.gradle.kts [hunk 1]
```diff
@@ -81,4 +97,13 @@ dependencies {
 
 
     // Serialization
+
+    //apollo
+    implementation("com.apollographql.apollo3:apollo-runtime:3.8.2")
```
Comment (michaelbui99) [inline app/build.gradle.kts:102]: Move to Dependencies.kt + Versions.kt🔥

File: app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/data/ApolloCountryClient.kt [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+            SimpleCountry(
+                it.name,
+                it.capital,
+                it.code)
+        } ?: emptyList()
+    }
+
+    override suspend fun getCountry(code: String): SimpleCountry {
+       val apolloClient = ApolloClient.Builder()
+            .serverUrl("https://countries.trevorblades.com/")
```
Comment (michaelbui99) [inline app/src/main/java/io/github/viabachelora23michaelkutaibakasper/bprapp/data/ApolloCountryClient.kt:27]: We probably want to be able to pass the URL in the function itself in the actual future client implementations, e.g. getCountry(code: String, url: String), getCountries(url: String) 🚀

File: build.gradle.kts [hunk 3]
```diff
@@ -7,10 +7,15 @@ buildscript {
     }
     dependencies {
         classpath (`secrets-gradle-plugin`())
+        classpath("com.android.tools.build:gradle:7.1.0")
```
Comment (michaelbui99) [inline build.gradle.kts:10]: Dependencies.kt + Versions.kt 🚀

**Review by michaelbui99 — CHANGES_REQUESTED** [review]
🔥 🚀 🌕

**Comment by Kasper-S-Jensen** [comment]
> Very nice, only minor changes needed. You can either make the changes or just create TODO's and merge it 🔥 🚀 🌕 

?!?!!?!!?

**Review by Kutaiba-Nezar-Kashmar — APPROVED** [review]
🔥🔥

**Pull request merged**